plugins/outputs/stackdriver/stackdriver.go

```diff
@@ -151,25 +171,49 @@ func (s *Stackdriver) Write(metrics []telegraf.Metric) error {
 			}
 
 			// Prepare time series.
-			timeSeries = append(timeSeries,
-				&monitoringpb.TimeSeries{
-					Metric: &metricpb.Metric{
-						Type:   path.Join("custom.googleapis.com", s.Namespace, m.Name(), f.Key),
-						Labels: getStackdriverLabels(m.TagList()),
-					},
-					MetricKind: metricKind,
-					Resource: &monitoredrespb.MonitoredResource{
-						Type:   s.ResourceType,
-						Labels: s.ResourceLabels,
-					},
-					Points: []*monitoringpb.Point{
-						dataPoint,
-					},
-				})
+			timeSeries := &monitoringpb.TimeSeries{
+				Metric: &metricpb.Metric{
+					Type:   path.Join("custom.googleapis.com", s.Namespace, m.Name(), f.Key),
+					Labels: getStackdriverLabels(m.TagList()),
+				},
+				MetricKind: metricKind,
+				Resource: &monitoredrespb.MonitoredResource{
+					Type:   s.ResourceType,
+					Labels: s.ResourceLabels,
+				},
+				Points: []*monitoringpb.Point{
+					dataPoint,
+				},
+			}
+
+			buckets.Add(m, f, timeSeries)
 		}
+	}
 
-		if len(timeSeries) < 1 {
-			continue
+	// process the buckets in order
+	keys := make([]uint64, 0, len(buckets))
+	for k := range buckets {
+		keys = append(keys, k)
+	}
+	sort.Slice(keys, func(i, j int) bool { return keys[i] < keys[j] })
```

Comment:
Why do we need to sort?

Reply:
In order to have a predictable output (request order) since maps ordering is not guaranteed.

Reply:
Okay